### ISSUE TITLE
 I couldn't run

### ISSUE THREAD

**Issue by barancanatbas** [zentures/encoding#8]
hello, I'm trying to replace the fastPFor java code with golang, but when the array I sent in my java code is outputting, I'm getting an error over golang.
![image](https://user-images.githubusercontent.com/64152397/216925779-e8a48dc6-ae5e-481b-b1b0-834d257f7f8c.png)

The error I get is: fastpfor/Compression Length = 0. No work done.

What exactly could be the reason, thanks in advance.

**Comment by lemire** [comment]
It is not my project but it appears that you are asking it to store the compressed result in an empty slice (out). It may not have the room to write anything!

**Comment by barancanatbas** [comment]
I don't think so, I tried with "out := make([]int32, 100)" but still got the same error. 
`inlength = encoding.FloorBy(inlength, DefaultBlockSize)
	if inlength == 0 {
		return errors.New("fastpfor/Compress: inlength = 0. No work done.")
	}`


I have 23k values ​​in my input array. I think there is a problem here

**Comment by lemire** [comment]
> I have 23k values ​​in my input array

In your example, you appear to have a single value in the array.

This being said, `fastpfor.New()` is not meant to be used alone. It is a block-oriented compressor. I don't know what the exact configuration is, in this project, but it probably will compress blocks of (say) 256 values. You can't compress a single integer with it. So you need to pair `fastpfor` with a code that can handle the tail. You can do so by following the conventions of this project as follows...

```
	codec =  composition.New(fastpfor.New(), variablebyte.New())
	err := codec.Compress(dd, inpos, len(dd), out, outpos)
```

Does that help?

Note that I am not the maintainer of this project.

**Comment by barancanatbas** [comment]
```bash
$ go mod tidy

go: finding module for package github.com/dataence/bytebuffer
go: finding module for package github.com/willf/bitset
go: finding module for package code.google.com/p/snappy-go/snappy
go: found github.com/dataence/bytebuffer in github.com/dataence/bytebuffer v0.0.0-20131118020616-f1bbd176b4c1
go: found github.com/willf/bitset in github.com/willf/bitset v1.7.0
go: testler imports
        github.com/zentures/encoding/fastpfor tested by
        github.com/zentures/encoding/fastpfor.test imports
        github.com/dataence/encoding/generators imports
        github.com/willf/bitset: github.com/willf/bitset@v1.7.0: parsing go.mod:
        module declares its path as: github.com/bits-and-blooms/bitset
                but was required as: github.com/willf/bitset
```

I'm getting an error while downloading the package, is it because of me?
